rbac/management/workspace/view.py

```diff
@@ -140,7 +142,17 @@ def validate_workspace(self, request, action="create"):
             self.validate_required_fields(request, REQUIRED_PUT_FIELDS)
         if parent_id:
             validate_uuid(parent_id)
-            if not Workspace.objects.filter(id=parent_id, tenant=tenant).exists():
+            try:
+                if self._exceeds_depth_limit(parent_id, tenant):
+                    message = f"Workspaces may only nest {settings.WORKSPACE_HIERARCHY_DEPTH_LIMIT} levels deep."
+                    error = {"workspace": [message]}
+                    raise serializers.ValidationError(error)
+            except ObjectDoesNotExist:
                 message = f"Parent workspace '{parent_id}' doesn't exist in tenant"
                 error = {"workspace": [message]}
                 raise serializers.ValidationError(error)
+
+    def _exceeds_depth_limit(self, parent_id, tenant):
```

Comment:
In order to remove one query from `Workspace#ancestors` , it is possible to put this method (or some variant) on workspace model and call just existing `_ancestry_queryset` to get length. (without   `Workspace.objects.filter(id__in=ancestor_ids)` which is in `Workspace#ancestors`)

Reply:
@lpichler I've updated some of the ancestor/descendant logic and queries in #1596 - what would you think about doing something similar to this: https://github.com/RedHatInsights/insights-rbac/pull/1596/files#diff-c71184a4354243afa33f717d763b1170dc7d65fa60f738c90fefc60e4f7f4befR70 where we could have something for ancestry that would just return the ancestor leave ids (and itself), in which case we wouldn't need to add the additional count. 

In #1596 I note that at first I abstracted the CTE out to be reused in the `descendants` method, but it would add an additional filter query. I'd like feedback on that, which we might be able to also use here if it makes sense and we DRY it up a bit.

Reply:
Yes, I agree with the approach in #1596 and (we can do discuss it in  #1596 and use it here)